### PR TITLE
Proposal: Use `ValidationError` over `ValueError` in views

### PR DESCRIPTION
**Pull request by macobo**
This generates a nicer exception for our users (400 response instead of
500) and we cut down on 500 errors for things that are not expected to
work.

Example sentry error with someone reverse-engineering our api: https://sentry.io/organizations/posthog/issues/2482887379/?project=1899813&query=is%3Aunassigned+is%3Aunresolved

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [ ] Frontend/CSS is usable at 320px (iPhone SE) and decent at 360px (most phones)
- [ ] Breaking changes are backwards-compatible. Ensure old/new frontend requests work with new/old backends, and vice versa.

### CODE REVIEW

File: ee/clickhouse/models/action.py [hunk 1]
```diff
@@ -93,7 +94,7 @@ def format_entity_filter(entity: Entity, prepend: str = "action", filter_by_team
             action = Action.objects.get(pk=entity.id)
             entity_filter, params = format_action_filter(action, prepend=prepend, filter_by_team=filter_by_team)
         except Action.DoesNotExist:
-            raise ValueError("This action does not exist")
+            raise ValidationError("This action does not exist")
```
Comment (paolodamico) [inline ee/clickhouse/models/action.py:97]: One suggestion here is to add a `code` param to provide a machine-friendly code representation that can then help frontend do special handling (or better API processing for users who connect directly). Example:

```python
raise ValidationError("This action does not exist", code="does_not_exist")
```

would output:
```json
{
  "type": "validation_error",
  "code": "does_not_exist",
  "detail": "This action does not exist",
  "attr": null
}
```

File: posthog/models/filters/stickiness_filter.py [hunk 2]
```diff
@@ -36,11 +37,11 @@ def __init__(self, data: Optional[Dict[str, Any]] = None, request: Optional[Http
         super().__init__(data, request, **kwargs)
         team: Optional[Team] = kwargs.get("team", None)
         if not team:
-            raise ValueError("Team must be provided to stickiness filter")
+            raise ValidationError("Team must be provided to stickiness filter")
         self.team = team
         get_earliest_timestamp: Optional[Callable] = kwargs.get("get_earliest_timestamp", None)
         if not get_earliest_timestamp:
-            raise ValueError("Callable must be provided when date filtering is all time")
+            raise ValidationError("Callable must be provided when date filtering is all time")
```
Comment (paolodamico) [inline posthog/models/filters/stickiness_filter.py:44]: Hmm maybe this should be a `ValueError`? Doesn't seem like this is something the user can fix, if it's not set it's probably bad code.

File: posthog/queries/stickiness.py [hunk 3]
```diff
@@ -123,7 +124,7 @@ def stickiness_process_entity_type(target_entity: Entity, team: Team, filter: St
             base.filter_events(team.pk, filter, target_entity)
         )
     else:
-        raise ValueError("target entity must be action or event")
+        raise ValidationError("target entity must be action or event")
```
Comment (paolodamico) [inline posthog/queries/stickiness.py:127]: Nit: capitalized & period.

**Review by paolodamico — APPROVED** [review]
great stuff here!

File: posthog/models/filters/stickiness_filter.py (hunk 2 above)
Comment (macobo) [reply posthog/models/filters/stickiness_filter.py:44]: It's someone trying to reverse-engineer the API and not passing in the right arguments for the endpoint -> it falls under this umbrella.

**Review by Twixes — APPROVED** [review]
Theoretically DRF exceptions should only be thrown in DRF-derived classes, not places like our own queries… But this definitely is a UX improvement with minimum effort, so 🚢

**Pull request merged**